### ISSUE TITLE
Rule Change: add suggestions for `eqeqeq`

### ISSUE THREAD

**Issue by rakleed** [eslint/eslint#19638]
### What rule do you want to change?

eqeqeq

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
if (x == 42) { }
if ("" == text) { }
if (obj.getStuff() != undefined) { }

a == b
foo == true
bananas != 1
value == undefined
foo == null
```

### What does the rule currently do for this code?

Reporting an error.

### What will the rule do after it's changed?

Suggests changing to strict comparison.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

There was an earlier discussion at https://github.com/eslint/eslint/issues/4578, and at that time you decided to disable auto-fix for `eqeqeq` altogether.

Then, for some cases, you did [bring it back](https://github.com/eslint/eslint/pull/7389); for those cases in the documentation, the auto-fix works:

```js
typeof foo == 'undefined'
'hello' != 'world'
0 == 0
true == true
```

The last time there was a discussion about this, I understand there were no suggestions back then, but it seems that for the other cases, the suggestions would have worked fine.

**Comment by snitin315** [comment]
This makes sense to me, this rule should provide suggestions. @eslint/eslint-tsc WDYT?

I'll work on this if accepted.

**Comment by mdjermanovic** [comment]
Makes sense to me, too.  Marking as accepted.
